testing/python/metafunc.py

```diff
@@ -1339,6 +1359,25 @@ def test_it2(foo):
         reprec = pytester.inline_run("--collect-only")
         assert not reprec.getcalls("pytest_internalerror")
 
+    def test_parametrize_with_pathlib_object(self, pytester: Pytester) -> None:
+        """#8118"""
+        pytester.makepyfile(
+            """
+            import pathlib
+            import pytest
+
+            @pytest.mark.parametrize('file', [pathlib.Path("test1.txt")])
+            def test_runscript(file):
+                assert True
+        """
+        )
+        result = pytester.runpytest("-v", "-s")
+        result.stdout.fnmatch_lines(
+            [
+                "test_parametrize_with_pathlib_object.py::test_runscript[test1.txt] PASSED",
```

Comment:
given this test output, I'm not sure this should be merged.  it's now difficult (impossible?) to differentiate:

```python
@pytest.mark.parametrize('s', (Path('test.txt'), 'test.txt'))
```

Reply:
> given this test output, I'm not sure this should be merged. it's now difficult (impossible?) to differentiate:
> 
> ```python
> @pytest.mark.parametrize('s', (Path('test.txt'), 'test.txt'))
> ```

Do you mean differentiate solely based on names? Or to differentiate in the test output which was a string and which was a `Path` object?

For the former, it looks like `_pytest.python.idmaker` should ensure uniqueness in the ids here, running the example above with `@pytest.mark.parametrize('file', [pathlib.Path("test1.txt"), "test1.txt"])` the output contains:

```
collected 2 items                                                                                                                                             

example.py::test_runscript[test1.txt0] PASSED
example.py::test_runscript[test1.txt1] PASSED
```

But for the latter this won't help.

Reply:
it makes the pathlib entries look identical to their string counterparts -- which they're not

Reply:
> it makes the pathlib entries look identical to their string counterparts -- which they're not

Yep, fair point. Do you think there's a reasonable solution here (first ideas into my head: decorator the id with the class or just `"Path"`)? Or do you think the feature is not worth it?

Reply:
I'm not sure to be honest, maybe something like `Path:input.txt`? (though, iirc `:` is a reserved character in ids -- maybe `|`?)

Reply:
My opinion is that it's OK to just use the straight string without disambiguation (other than appending the sequence on conflict which pytest already does). For me the most natural ID for a path is just its string, and I don't think it would cause confusion in practice.

Reply:
Well, my opinion is that it's not :P